File: coresuite-dotnet-standard-vb/reportwriter/ContactListWithGroupByReport.vb
Imports ceTe.DynamicPDF
Imports ceTe.DynamicPDF.LayoutEngine
Imports coresuite_dotnet_reportdata_vb

Namespace coresuite_dotnet_standard_vb.reportwriter

    Public Class ContactListWithGroupByReport

        Public Shared Sub Run(outputPdfPath As String)


            ' Create the document's layout from a DLEX template
            Dim documentLayout As DocumentLayout = New DocumentLayout(Util.GetResourcePath("DLEX/ContactListWithGroupBy.dlex"))

            Dim layoutData As NameValueLayoutData = New NameValueLayoutData()
            layoutData.Add("Contacts", ContactListWithGroupByExampleData.Contacts)

            ' Layout the document using the parameters and set it's properties
            Dim document As Document = documentLayout.Layout(layoutData)
            document.Author = "DynamicPDF ReportWriter"
            document.Title = "Contact List With Group By Example"

            ' Outputs the document to a file
            document.Draw(outputPdfPath)
        End Sub
    End Class
End Namespace

File: coresuite-dotnet-core-console-cs/reportwriter/ContactListWithGroupByReport.cs
using ceTe.DynamicPDF;
using ceTe.DynamicPDF.LayoutEngine;

namespace coresuite_dotnet_core_console_cs.reportwriter
{
    public class ContactListWithGroupByReport
    {
        public static void Run(string outputPdfPath)
        {
            // Create the document's layout from a DLEX template
            DocumentLayout documentLayout = new DocumentLayout(Util.GetResourcePath("DLEX/ContactListWithGroupBy.dlex"));

            NameValueLayoutData layoutData = new NameValueLayoutData();
            layoutData.Add("Contacts", ContactListWithGroupByExampleData.Contacts);

            // Layout the document using the parameters and set it's properties
            Document document = documentLayout.Layout(layoutData);
            document.Author = "DynamicPDF ReportWriter";
            document.Title = "Contact List With Group By Example";

            // Outputs the document to a file
            document.Draw(outputPdfPath);
        }
    }
}